code-studio/src/js/videos.js

```diff
@@ -0,0 +1,312 @@
+    try {
+      window.sessionStorage.setItem('lastTab', selected);
+    } catch (exc) {
+      console.log('Caught exception in sessionStorage.setItem: ', exc);
+    }
+  };
+
+  var lastTab = window.sessionStorage.getItem('lastTab');
+  body.tabs({
+    event : 'click touchend',
+    activate: tabHandler,
+    create: tabHandler,
+    active: (lastTab !== null) ? lastTab : 0  // Set starting tab.
+  });
+
+  var download = $('<a/>').append($('<img src="/shared/images/download_button.png"/>'))
+      .addClass('download-video')
+      .attr('href', options.download);
+  var nav = $div.find('.ui-tabs-nav');
+  nav.append(download);
+
+  // Resize modal to fit constraining dimension.
+  var height = $(window).height() * widthRatio,
+      width = $(window).width() * heightRatio;
+
+  if (height * aspectRatio < width) {
+    $div.height(height);
+    $div.width(height * aspectRatio);
+  } else {
+    $div.height(width / aspectRatio);
+    $div.width(width);
+  }
+
+  // Standard css hack to center a div within the viewport.
+  $div.css({
+    top: '50%',
+    left: '50%',
+    marginTop: ($div.height() / -2) + 'px',
+    marginLeft: ($div.width() / -2) + 'px'
+  });
+
+  var divHeight = $div.innerHeight() - nav.outerHeight();
+  $(video).height(divHeight);
+
+  notesDiv.height(divHeight);
+
+  if (window.YT && window.YT.loaded) {
+    onYouTubeIframeAPIReady();
+  } else {
+    // Use the official YouTube IFrame Player API to load the YouTube video.
+    // Ref: https://developers.google.com/youtube/iframe_api_reference#Getting_Started
+    var tag = document.createElement('script');
+    tag.src = "https://www.youtube.com/iframe_api";
+    var firstScriptTag = document.getElementsByTagName('script')[0];
+    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+    // calls window.onYouTubeIframeAPIReady
+  }
+
+  dialog.show();
+
+  var videoModal = $('.video-modal');
+
+  videoModal.on("ended", function () {
+    dialog.hide();
+  });
+
+  // Raise an event that the dialog has been shown, in case anything needs to
+  // pause/respond to it.
+  var event = document.createEvent('Event');
+  event.initEvent('videoShown', true, true);
+  document.dispatchEvent(event);
+
+  // Don't add fallback player if a video modal has closed
+  var shouldStillAdd = true;
+  videoModal.one('hidden.bs.modal', function(){
+    shouldStillAdd = false;
+  });
+
+  setupVideoFallback(options, $div.width(), divHeight, function(){
+    return shouldStillAdd;
+  });
+};
+
+/**
+ * When hidden quickly after being shown, the videojs flash player progress tracker
+ * starts looping out read property 'length' of undefined errors. Since we don't currently
+ * use this progress tracking, disable it
+ */
+function stopTrackingVideoJSProgress() {
+  if (!$('.video-js').length) {
+    return;
+  }
+
+  var fallbackPlayer = videojs($('.video-js')[0]);
+  if (fallbackPlayer.stopTrackingProgress) {
+    fallbackPlayer.stopTrackingProgress();
+  }
```

Comment:
ah, if an unminified dev version is also provided then yeah, it might be preferable to link to that version if possible. Maybe something like `require('video.js/dist/video-js/video.dev')` instead of `require('video.js')`?